toolkit/tools/internal/randomization/randomization_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package randomization
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"microsoft.com/pkggen/internal/logger"
+)
+
+const (
+	normalString    = "abcde12345"
+	multiByteString = "世界您好IsHelloWorld"
+	emojiString     = "1🤷‍♂️2@3( •_•)>⌐■~■ab~52(⌐■_■)67👩‍💻"
+)
+
+func TestMain(m *testing.M) {
+	testResult := 0
+
+	logger.InitStderrLog()
+	testResult = m.Run()
+
+	os.Exit(testResult)
+}
+
+func TestRandomNoNormalStringError(t *testing.T) {
+	var (
+		inputString = normalString
+		length      = 100
+	)
+	randomString, err := RandomString(length, inputString)
+	assert.NoError(t, err)
+	// We need to count runes, not
+	assert.Equal(t, length, len([]rune(randomString)))
+
+	fmt.Println(randomString)
+	fmt.Printf("len: %d\n", len(randomString))
+
+	// Heuristic to check we are getting a "roughly" random distribution:
+	// Generate random strings, check if each character is used at least an (average / 2) number of times at some point during the trials
+	// ie 100/10 = 10, 10/2 = 5, make sure every character shows up at least 5 times during the repeated trials
+	goodReturn := make(map[rune]bool)
+	for trial := 0; trial < 10; trial++ {
```

Comment:
>for trial := 0; trial < 10; trial++ { [](start = 1, length = 37)

Are multiple runs needed instead of one, longer random string?

Reply:
Just do one test now, the check function still validates the strings are long enough.

---
In reply to: [550340220](https://github.com/microsoft/CBL-Mariner/pull/479#discussion_r550340220) [](ancestors = 550340220)